assets/js/modules/adsense/components/setup/v2/SetupAccountSite/Ready.js

```diff
@@ -0,0 +1,155 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useContext, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import ViewContextContext from '../../../../../../components/Root/ViewContextContext';
+import { trackEvent } from '../../../../../../util';
+import { MODULES_ADSENSE } from '../../../../datastore/constants';
+import SetupAccountSiteUI from '../common/SetupAccountSiteUI';
+const { useSelect, useDispatch } = Data;
+
+export default function Ready( { site, finishSetup } ) {
+	const [
+		acknowledgedDisabledAutoAds,
+		setAcknowledgedDisabledAutoAds,
+	] = useState( false );
+
+	const viewContext = useContext( ViewContextContext );
+
+	const existingTag = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getExistingTag()
+	);
+
+	const enableAutoAdsURL = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getServiceAccountSiteAdsPreviewURL()
+	);
+
+	const isDoingSubmitChanges = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).isDoingSubmitChanges()
+	);
+
+	const { completeSiteSetup, completeAccountSetup } = useDispatch(
+		MODULES_ADSENSE
+	);
+
+	const enableAutoAdsHandler = useCallback(
+		( event ) => {
+			event.preventDefault();
+			trackEvent( `${ viewContext }_adsense`, 'enable_auto_ads' );
+			global.open( enableAutoAdsURL, '_blank' );
+		},
+		[ enableAutoAdsURL, viewContext ]
+	);
+
+	const acknowledgeDisabledAutoAdsHandler = useCallback(
+		async ( event ) => {
+			event.preventDefault();
+			await trackEvent( `${ viewContext }_adsense`, 'disable_auto_ads' );
+			setAcknowledgedDisabledAutoAds( true );
+		},
+		[ viewContext ]
+	);
+
+	const continueSetupHandler = useCallback( async () => {
+		if ( isDoingSubmitChanges ) {
+			return;
+		}
+
+		const successSiteSetupCompletion = await completeSiteSetup();
+		const successAccountSetupCompletion = await completeAccountSetup();
+		if (
+			successSiteSetupCompletion &&
+			successAccountSetupCompletion &&
+			typeof finishSetup === 'function'
+		) {
+			finishSetup();
+		}
+	}, [
+		isDoingSubmitChanges,
+		finishSetup,
+		completeSiteSetup,
+		completeAccountSetup,
+	] );
+
+	const uiProps = {};
+	if ( ! site.autoAdsEnabled && ! acknowledgedDisabledAutoAds ) {
+		uiProps.heading = __(
+			'Enable auto ads for your site',
+			'google-site-kit'
+		);
+
+		uiProps.description = existingTag
+			? __(
+					'Site Kit recommends enabling auto ads. If your existing AdSense setup relies on individual ad units, you can proceed without enabling auto ads.',
+					'google-site-kit'
+			  )
+			: __(
+					'To start serving ads via Site Kit, you need to activate auto ads first. Go to AdSense and enable auto ads for your site.',
+					'google-site-kit'
+			  );
```

Comment:
Let's move this to the existingTag condition to avoid having two similar conditions:

```js
if ( existingTag ) {
    uiProps.secondaryButton = { ... };
    uiProps.description = __( ... );
} else {
    uiProps.description = __( ... );
}
```